digicampipe/utils/disp.py

```diff
@@ -0,0 +1,290 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib as mpl
+from tqdm import trange
+
+
+def disp_eval(A, width, length, cog_x, cog_y,
+              x_offset, y_offset, psi, skewness,
+              size, leakage2, method):
+
+    parvals = A.valuesdict()
+
+    # (Lessard et al., 2001)
+    if method == 1:
+        disp_comp = parvals['A0']*(1 - width/length)
+
+    # eq 2.10 in Lopez Coto, VHE gamma-ray observations of pulsar
+    # wind nebuale ..., but better reference is: (Domingo-Santamaria+, 2005)
+    elif method == 2:
+        A = (parvals['A0'] + parvals['A1'] * np.log10(size)
+             + parvals['A2'] * np.log10(size)**2)
+        B = (parvals['A3'] + parvals['A4'] * np.log10(size)
+             + parvals['A5'] * np.log10(size)**2)
+        eta = (parvals['A6'] + parvals['A7'] * np.log10(size)
+               + parvals['A8'] * np.log10(size)**2)
+        disp_comp = A + B * (width / (length + eta * leakage2))
+
+    # just some test
+    elif method == 3:
+        disp_comp = parvals['A0'] + parvals['A1'] * length/width
+
+    # Kranich and Stark, ICRC 2003
+    elif method == 4:
+        disp_comp = (parvals['A0']
+                     * (1 - width / (length * (1 + parvals['A1'] * leakage2))))
+    # (Luke Riley St Marie 2014) <-- simplified Domingo-Santamaria
+    elif method == 5:
+        fraction = width / (length + parvals['A2']
+                            * leakage2 * np.log10(size))
+        disp_comp = (np.log10(size)
+                     * (parvals['A0'] + parvals['A1'] * (1 - fraction)))
+
+    x_source_comp0 = cog_x + disp_comp*np.cos(psi)  # Two possible solutions
+    y_source_comp0 = cog_y + disp_comp*np.sin(psi)  #
+    x_source_comp1 = cog_x - disp_comp*np.cos(psi)  #
+    y_source_comp1 = cog_y - disp_comp*np.sin(psi)  #
+
+    # Selection of one specific solution according to skewness
+    # - head of the shower (close to the source) gives more signal
+    # - if skewness > 0, head is closer to the center of FOV than tail
+    x_source_comp = []
```

Comment:
would that do the work ?: 

```python
x_source_comp = np.zeros(disp_comp.shape)
y_source_comp = np.zeros(disp_comp.shape)


skewness_is_positive = skewness > 0

x_source_comp[skewness_is_positive] = x_source_comp1
x_source_comp[~skewness_is_positive] = x_source_comp0
y_source_comp[skewness_is_positive] = y_source_comp1
y_source_comp[~skewness_is_positive] = y_source_comp0

residuals = np.array([x_offset-x_source_comp, y_offset-y_source_comp])
```

Reply:
Thanks! It works, I just had to use x_source_comp1[skewness_is_positive], because x_source_comp and x_source_comp1 have the same length.